pkg/liqo-controller-manager/authentication/identity-controller/identity_controller.go

```diff
@@ -0,0 +1,151 @@
+		return ctrl.Result{}, err
+	}
+
+	// Ensure the kubeconfig secret.
+	secret, err := r.ensureKubeconfigSecret(ctx, &identity)
+	if err != nil {
+		klog.Errorf("unable to ensure kubeconfig secret for identity %q: %v", req.NamespacedName, err)
+		return ctrl.Result{}, err
+	}
+
+	// Update the identity status to reference the kubeconfig secret.
+	if err := r.handleIdentityStatus(ctx, &identity, client.ObjectKeyFromObject(secret)); err != nil {
+		klog.Errorf("unable to update identity %q status: %v", req.NamespacedName, err)
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *IdentityReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&authv1alpha1.Identity{}).
+		Owns(&corev1.Secret{}).
+		Complete(r)
+}
+
+func (r *IdentityReconciler) ensureKubeconfigSecret(ctx context.Context, identity *authv1alpha1.Identity) (*corev1.Secret, error) {
+	// Get the private Key encoded in PEM format.
+	privateKey, _, err := authentication.GetClusterKeysPEM(ctx, r.Client, r.liqoNamespace)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get cluster keys: %w", err)
+	}
+
+	// If it's an identity used for the CRD Replicator (type: ControlPlane) we set the remote namespace as default
+	// for the kubeconfig.
+	var namespace *string
+	if identity.Spec.Type == authv1alpha1.ControlPlaneIdentityType {
+		namespace = identity.Spec.Namespace
+	}
+
+	// Create or update the secret containing the kubeconfig.
+	kubeconfigSecret := forge.KubeconfigSecret(identity)
+	op, err := controllerutil.CreateOrUpdate(ctx, r.Client, kubeconfigSecret, func() error {
+		if err := forge.MutateKubeconfigSecret(kubeconfigSecret, identity, privateKey, namespace); err != nil {
+			return err
+		}
+		return controllerutil.SetOwnerReference(identity, kubeconfigSecret, r.Scheme)
+	})
+	if err != nil {
+		return nil, fmt.Errorf("unable to create or update kubeconfig secret: %w", err)
+	}
+
+	if op != controllerutil.OperationResultNone {
+		klog.Infof("kubeconfig secret %q %s", client.ObjectKeyFromObject(kubeconfigSecret), op)
+		r.eventRecorder.Event(kubeconfigSecret, "Normal", "KubeconfigEnsured", "Kubeconfig secret ensured")
+	}
+
+	return kubeconfigSecret, nil
+}
+
+// handleIdentityStatus updates the identity status to reference the kubeconfig secret.
+func (r *IdentityReconciler) handleIdentityStatus(ctx context.Context, identity *authv1alpha1.Identity, secretNsName types.NamespacedName) error {
+	var kubeconfigSecret corev1.Secret
+	if err := r.Get(ctx, secretNsName, &kubeconfigSecret); err != nil {
+		return err
+	}
+
+	identity.Status.KubeconfigSecretRef = &corev1.ObjectReference{
+		Namespace: kubeconfigSecret.Namespace,
+		Name:      kubeconfigSecret.Name,
+		Kind:      kubeconfigSecret.Kind,
+		UID:       kubeconfigSecret.UID,
+	}
+
+	if err := r.Status().Update(ctx, identity); err != nil {
+		return err
+	}
```

Comment:
I would keet the update in the reconcile function. I think is cleaner to modify the object in the subfunctions, but save it only in the main one